src/main/java/alpha/FileHandler.java

```diff
@@ -0,0 +1,192 @@
+        if (isCreated) {
+            return taskList;
+        }
+        try {
+            Scanner sc = new Scanner(alphaTxt);
+            while (sc.hasNext()) {
+                String input = sc.nextLine();
+                String[] splitInput = input.split("\\|");
+                String type = splitInput[0];
+                String checked = splitInput[1];
+                String description = splitInput[2].trim();
+                Task task;
+                if (type.equals("T ")) {
+                    task = ToDo.createToDo(description);
+                } else if (type.equals("D ")) {
+                    task = Deadline.makeDeadline(description, splitInput[3]);
+                } else {
+                    task = Event.makeEvent(description, splitInput[3], splitInput[4]);
+                }
+                if (checked.equals("X")) {
+                    assert task != null;
+                    task.mark();
+                }
+                taskList.add(task);
+            }
+            return taskList;
+        } catch (FileNotFoundException e) {
+            System.out.println("The file could not be found");
+            return taskList;
+        } catch (Exception e) { // The code should never reach this stage since everything saved to the file should be correctly formatted.
+            return taskList;
+        }
+    }
+
+    /**
+     * Saves a task to the file.
+     *
+     * @param task The task to add to the file.
+     */
+    // Saves Task to file whenever a Task is added
+    public void saveToFile(Task task) {
+        try {
+            writer = new FileWriter(alphaTxt, true);
+            if (alphaTxt.length() != 0) {
+                writer.write(System.lineSeparator());
+            }
+            if (task instanceof Deadline) {
+                writer.write("D |" + task.getStatusIcon() + "| " + task.getDescription() + " | "
+                        + ((Deadline) task).getDateBy() + " " + ((Deadline) task).getTimeBy());
+            } else if (task instanceof ToDo) {
+                writer.write("T |" + task.getStatusIcon() + "| " + task.getDescription());
+            } else {
+                writer.write("E |" + task.getStatusIcon() + "| " + task.getDescription() + " | " + ((Event) task).getStart()
+                        + " | " + ((Event) task).getEnd());
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("I/O Exception");
+        }
+    }
+
+    /**
+     * Checks or unchecks a task in the file depending on the second boolean parameter.
+     *
+     * @param index The index of the task to be marked or unmarked.
+     * @param check Boolean of whether the task is to be checked or unchecked.
+     */
+    // Checks or unchecks a task in the file. Does so by creating a temp file that copies everything over
+    // except the task that is being checked/unchecked.
+    public void checkOrUncheck(int index, boolean check) {
```

Comment:
Any reason why you used a temp file instead of just overwriting the current file? Not sure if this will affect the efficiency